File: src/main/java/com/manage/interior_inventory/service/SupplierInvoiceService.java
package com.manage.interior_inventory.service;

import com.manage.interior_inventory.dto.finance.invoice.SupplierInvoiceUploadResponse;
import com.manage.interior_inventory.dto.finance.invoice.UpdateInvoiceAmountRequest;

import org.springframework.web.multipart.MultipartFile;
import java.util.List;

public interface SupplierInvoiceService {

    /** 上傳 PDF → 解析 → 比對 → 寫入 DB → 回傳比對結果 */
    SupplierInvoiceUploadResponse uploadAndParse(Long projectId, MultipartFile file);

    /** 查詢單一對帳單的完整明細（含批次分組） */
    SupplierInvoiceUploadResponse getDetail(Long invoiceId);

    /** 查詢某案件的所有對帳單摘要列表 */
    List<SupplierInvoiceUploadResponse> listByProject(Long projectId);

    SupplierInvoiceUploadResponse updateAmounts(Long invoiceId, UpdateInvoiceAmountRequest req);
}

File: frontend/src/services/supplierInvoiceService.ts
import apiClient from '@/lib/apiClient'
import type { SupplierInvoiceResponse, UpdateInvoiceAmountRequest } from '@/types/finance'

const BASE = '/finance/supplier-invoices'

export const supplierInvoiceService = {

    /** POST /api/finance/supplier-invoices/upload
     *  form-data: projectId, file
     *  → 解析比對結果
     */
    uploadAndParse: (
        projectId: number,
        file: File
    ): Promise<SupplierInvoiceResponse> => {
        const form = new FormData()
        form.append('projectId', String(projectId))
        form.append('file', file)
        return apiClient
            .post(`${BASE}/upload`, form, {
                headers: { 'Content-Type': 'multipart/form-data' },
            })
            .then(r => r.data)
    },

    /** GET /api/finance/supplier-invoices/{invoiceId}
     *  → 單一對帳單完整明細
     */
    getDetail: (invoiceId: number): Promise<SupplierInvoiceResponse> =>
        apiClient.get(`${BASE}/${invoiceId}`).then(r => r.data),

    /** GET /api/finance/supplier-invoices/project/{projectId}
     *  → 案件所有歷史對帳單列表
     */
    listByProject: (projectId: number): Promise<SupplierInvoiceResponse[]> =>
        apiClient.get(`${BASE}/project/${projectId}`).then(r => r.data),

    /** PATCH /api/finance/supplier-invoices/{invoiceId}/amounts
     *  人工修正應收總額、現金扣款、付現應收
     */
    updateAmounts: (
        invoiceId: number,
        data: UpdateInvoiceAmountRequest
    ): Promise<SupplierInvoiceResponse> =>
        apiClient
            .patch(`${BASE}/${invoiceId}/amounts`, data)
            .then(r => r.data),

}